src/main/java/seedu/address/logic/commands/module/ModuleEditCommand.java

```diff
@@ -54,18 +60,39 @@ public ModuleEditCommand(Index index, ModuleEditCommand.EditModuleDescriptor edi
 
         this.index = index;
         this.editModuleDescriptor = new ModuleEditCommand.EditModuleDescriptor(editModuleDescriptor);
+        this.moduleCode = null;
+    }
+
+    public ModuleEditCommand(ModuleCode moduleCode, ModuleEditCommand.EditModuleDescriptor editModuleDescriptor) {
+        requireNonNull(moduleCode);
+        requireNonNull(editModuleDescriptor);
+
+        this.index = null;
+        this.moduleCode = moduleCode;
+        this.editModuleDescriptor = editModuleDescriptor;
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
         List<Module> lastShownList = model.getFilteredModuleList();
+        Module moduleToEdit = null;
 
-        if (index.getZeroBased() >= lastShownList.size()) {
+        if (index != null && index.getZeroBased() >= lastShownList.size()) {
             throw new CommandException(Messages.MESSAGE_INVALID_MODULE_DISPLAYED_INDEX);
+        } else if (index != null) {
+            moduleToEdit = lastShownList.get(index.getZeroBased());
+        } else { // index == null
+            assert moduleCode != null;
+            for (Module mod : lastShownList) {
+                if (mod.getModuleCode().equals(moduleCode)) {
+                    moduleToEdit = mod;
+                    break;
+                }
+            }
         }
 
-        Module moduleToEdit = lastShownList.get(index.getZeroBased());
+        assert moduleToEdit != null;
```

Comment:
I think you can throw exception instead when the module code does not exist.

Reply:
yeah i somehow thought the exception had been caught somewhere else
and then realized nope